### PR TITLE
Add default case for non-member types in assert_type_is

### LINKED ISSUE #23346: idlharness.js's assert_type_is needs to be taught about each interface
As noted in e.g. https://github.com/web-platform-tests/wpt/pull/23323#issuecomment-621677647 the idlharness assertion `assert_type_is()` needs to be explicitly made aware of interfaces like Float32Array and DataView. If an interface isn't listed, it leads to unexpected failures. 

We likely haven't noticed this much because most idlharness tests are boilerplate and don't provide sample values. Even after the above PR, grepping for the error in blink's output, I found some geometry-related fails: `" Unrecognized type"` --> DOMRect, DOMRectReadOnly, DOMPointReadOnly, DOMMatrix

Can we update `assert_type_is()` to have `value instanceof self[type]` as the default case? (Honest question: I didn't look deeply enough to see what the downsides are)

cc: @foolip

### PR DESCRIPTION
**Pull request by stephenmcgruer**
Fixes https://github.com/web-platform-tests/wpt/issues/23346

### CODE REVIEW

**Comment by stephenmcgruer** [comment]
Stability check failure was Firefox timing out. Results from wpt.fyi suggest no regressions, which is reasonable minimum bar.

Josh, Domenic - wdyt of this as a fix for https://github.com/web-platform-tests/wpt/issues/23346 ?

File: resources/idlharness.js [hunk 1]
```diff
@@ -1209,29 +1209,17 @@ IdlArray.prototype.assert_type_is = function(value, type)
...
-        case "Function":
-            assert_true(value instanceof self[type], "wrong type: not a " + type);
-            return;
-
         case "object":
             assert_in_array(typeof value, ["object", "function"], "wrong type: not object or function");
             return;
     }
 
+    // This is a catch-all for basic types (e.g. Int8Array). If a type needs
```
Comment (domenic) [inline resources/idlharness.js:1217]: I'd suggest

> This is a catch-all for any IDL type name which follows JS class semantics. This includes some non-interface IDL types (e.g. Int8Array, Function, ...), as well as any interface types that are not in the IDL that is fed to the harness. If an IDL type does not follow JS class semantics, then it should go in the switch statement above. If an IDL type needs full checking, then the test should include it in the IDL it feels to the harness.

**Review by domenic — APPROVED** [review]
The logic here LGTM. And if this causes no regressions then I'm happy to approve.

I worry a bit that without this, people will often forget to input important dependent IDL into their testharness tests. But probably the benefit in simple cases outweighs that.

**Review by inexorabletash — APPROVED** [review]
Hmmm... Is the set of intrinsic types from WebIDL enumerated anywhere?

For example, I see "Function" here but it's not listed anywhere in webidl2.js's source. It's a defined type in WebIDL that "coincidentally" has the same name in JS.

What I'm worried about: say someone writes an IDL with `RegExp` - a JS type but not a WebIDL type. Would that get caught by idlharness before (giving an error), but slip through now?

Approving though since this assert function is *not* where such things should get caught.

**Comment by inexorabletash** [comment]
Following up to my previous comment: looks like idlharness indeed doesn't validate that types are known. Having a function return any of (`Function`, `VoidFunction`, `NoSuchThing`, `RegExp`) don't cause errors. Plenty of open issues against idlharness already, though.

Still LGTM, thanks!

File: resources/idlharness.js (hunk 1 above)
Comment (stephenmcgruer) [reply resources/idlharness.js]: Done.

**Comment by stephenmcgruer** [comment]
Ok. I'm a bit hesitant here because the experts (y'all) are unsure (and I know so very little). But then again, experts also approved this so I guess its good :D.

Admin-merging because Firefox stability check is timing out due to number of tests affected.

**Pull request merged**